felix/fv/routing_test.go

```diff
@@ -0,0 +1,1086 @@
+						return out
+					}, "60s", "500ms").Should(ContainSubstring(mtuStr))
+				}
+
+				// Change the host device's MTU, and expect the IPIP device to be updated.
+				for _, felix := range felixes {
+					Eventually(func() error {
+						_, err := felix.ExecOutput("ip", "link", "set", "eth0", "mtu", "1400")
+						return err
+					}, "10s", "100ms").Should(BeNil())
+				}
+
+				// MTU should be auto-detected, and updated to the host MTU minus 20 bytes overhead IPIP.
+				mtuStr = "mtu 1380"
+				mtuValue := "1380"
+				for _, felix := range felixes {
+					// Felix checks host MTU every 30s
+					Eventually(func() string {
+						out, _ := felix.ExecOutput("ip", "-d", "link", "show", dataplanedefs.IPIPIfaceName)
+						return out
+					}, "60s", "500ms").Should(ContainSubstring(mtuStr))
+
+					// And expect the MTU file on disk to be updated.
+					Eventually(func() string {
+						out, _ := felix.ExecOutput("cat", "/var/lib/calico/mtu")
+						return out
+					}, "30s", "100ms").Should(ContainSubstring(mtuValue))
+				}
+
+				// Explicitly configure the MTU.
+				felixConfig := api.NewFelixConfiguration() // Create a default FelixConfiguration
+				felixConfig.Name = "default"
+				mtu := 1300
+				felixConfig.Spec.IPIPMTU = &mtu
+				_, err := client.FelixConfigurations().Create(context.Background(), felixConfig, options.SetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				// Expect the settings to be changed on the device.
+				for _, felix := range felixes {
+					// Felix checks host MTU every 30s
+					Eventually(func() string {
+						out, _ := felix.ExecOutput("ip", "-d", "link", "show", dataplanedefs.IPIPIfaceName)
+						return out
+					}, "60s", "500ms").Should(ContainSubstring("mtu 1300"))
+				}
+			})
+
+			Context("external nodes configured", func() {
+				var externalClient *containers.Container
+
+				BeforeEach(func() {
+					externalClient = infrastructure.RunExtClient("ext-client")
+
+					Eventually(func() error {
+						err := externalClient.ExecMayFail("ip", "tunnel", "add", "tunl0", "mode", "ipip")
+						if err != nil && strings.Contains(err.Error(), "SIOCADDTUNNEL: File exists") {
+							return nil
+						}
+						return err
+					}).Should(Succeed())
+
+					externalClient.Exec("ip", "link", "set", "tunl0", "up")
+					externalClient.Exec("ip", "addr", "add", "dev", "tunl0", "10.65.222.1")
+					externalClient.Exec("ip", "route", "add", "10.65.0.0/24", "via",
+						tc.Felixes[0].IP, "dev", "tunl0", "onlink")
+				})
+
+				JustAfterEach(func() {
+					if CurrentGinkgoTestDescription().Failed {
+						externalClient.Exec("ip", "r")
+						externalClient.Exec("ip", "l")
+						externalClient.Exec("ip", "a")
+					}
+				})
+
+				AfterEach(func() {
+					externalClient.Stop()
+				})
+
+				It("should allow IPIP to external client if it is in ExternalNodesCIDRList", func() {
+					if ipipMode == api.IPIPModeNever {
+						Skip("external nodes is not applicable to no encap routing")
+					}
```

Comment:
This test is about accepting tunnel traffic from external nodes: https://github.com/projectcalico/calico/blob/a9752ecb4f2dae41deef9ea967363ef188fc3126/api/pkg/apis/projectcalico/v3/felixconfig.go#L531

We insert a rule to accept tunnel traffic from the allowed external nodes 
here IPIP and VXLAN: https://github.com/projectcalico/calico/blob/a9752ecb4f2dae41deef9ea967363ef188fc3126/felix/rules/static.go#L249

but it's not relevant to no encap.